Formula/grafana-agent.rb

```diff
@@ -0,0 +1,98 @@
+class GrafanaAgent < Formula
+  desc "Exporter for Prometheus Metrics, Loki Logs, and Tempo Traces"
+  homepage "https://grafana.com/docs/agent/"
+  url "https://github.com/grafana/agent/archive/refs/tags/v0.21.2.tar.gz"
+  sha256 "a9deceaf1c09f19e5f44204e7d7c755dc24a5dd5a532c76c0524e762a695b590"
+  license "Apache-2.0"
+
+  depends_on "go" => :build
+
+  on_linux do
+    depends_on "systemd" => :build
+  end
+
+  def install
+    ldflags = %W[
+      -X github.com/grafana/agent/pkg/build.Branch=HEAD
+      -X github.com/grafana/agent/pkg/build.Version=v#{version}
+      -X github.com/grafana/agent/pkg/build.Revision=v#{version}
+      -X github.com/grafana/agent/pkg/build.BuildUser=Homebrew
+      -X github.com/grafana/agent/pkg/build.BuildDate=#{DateTime.now.rfc3339}
+    ]
+    system "go", "build", "-ldflags", ldflags.join(" "), "-trimpath",
+           "-o", bin/"grafana-agent", "./cmd/agent"
+    system "go", "build", "-ldflags", ldflags.join(" "), "-trimpath",
+           "-o", bin/"grafana-agentctl", "./cmd/agentctl"
+
+    (bin/"grafana_agent_brew_services").write <<~EOS
+      #!/bin/bash
+      exec #{bin}/grafana-agent $(<#{etc}/grafana-agent.args)
+    EOS
+
+    (buildpath/"grafana-agent.args").write <<~EOS
+      -config.file #{etc}/grafana-agent.yml
+      -prometheus.wal-directory #{var}/grafana-agent
+    EOS
+
+    (buildpath/"grafana-agent.yml").write <<~EOS
+      server:
+        log_level: info
+        http_listen_address: "127.0.0.1"
+        http_listen_port: 12345
+        grpc_listen_address: "127.0.0.1"
+        grpc_listen_port: 9095
+
+      metrics:
+        global:
+          scrape_interval: 15s
+
+      integrations:
+        node_exporter:
+          enabled: true
+    EOS
+    etc.install "grafana-agent.args", "grafana-agent.yml"
```

Comment:
We should not construct our own configuration and binaries. We only ship what upstream ships.

Reply:
Understood! Will fix.